.github/workflows/integration.yml

```diff
@@ -0,0 +1,272 @@
+name: Integration
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - stable*
+
+env:
+  APP_NAME: notifications
+
+jobs:
+  sqlite:
+    runs-on: ubuntu-latest
+
+    strategy:
+      # do not stop on another job's failure
+      fail-fast: false
+      matrix:
+        php-versions: ['7.4']
+        databases: ['sqlite']
+        server-versions: ['master']
+
+    name: integration-php${{ matrix.php-versions }}-${{ matrix.databases }}
+
+    steps:
+      - name: Checkout server
+        uses: actions/checkout@v2
+        with:
+          repository: nextcloud/server
+          ref: ${{ matrix.server-versions }}
+
+      - name: Checkout submodules
+        shell: bash
+        run: |
+          auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+          git submodule sync --recursive
+          git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+
+      - name: Checkout app
+        uses: actions/checkout@v2
+        with:
+          path: apps/${{ env.APP_NAME }}
+
+      - name: Set up php ${{ matrix.php-versions }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: mbstring, iconv, fileinfo, intl, sqlite, pdo_sqlite
+          tools: phpunit:8.5.2
```

Comment:
Maybe a bit wider? `phpunit:8` ? If available :thinking:

Reply:
copy pasta from somewhere, also the behat version needs to match.
I will try to get rid of the custom version with https://github.com/nextcloud/notifications/pull/901  when it works